pkg/apis/azure/storage/v1alpha1/types.go

```diff
@@ -0,0 +1,227 @@
+
+	// PublicAccessType
+	PublicAccessType azblob.PublicAccessType `json:"publicAccessType,omitempty"`
+
+	// AccountRef reference to azure storage account object
+	AccountRef corev1.LocalObjectReference `json:"accountRef"`
+
+	// ConnectionSecretNameOverride to generate connection secret with specific name
+	ConnectionSecretNameOverride string `json:"connectionSecretNameOverride,omitempty"`
+
+	ClaimRef *corev1.ObjectReference `json:"claimRef,omitempty"`
+	ClassRef *corev1.ObjectReference `json:"classRef,omitempty"`
+
+	// ReclaimPolicy identifies how to handle the cloud resource after the deletion of this type
+	ReclaimPolicy corev1alpha1.ReclaimPolicy `json:"reclaimPolicy,omitempty"`
+}
+
+// ContainerStatus sub-resource for Container object
+type ContainerStatus struct {
+	corev1alpha1.ConditionedStatus
+	corev1alpha1.BindingStatusPhase
+	ConnectionSecretRef corev1.LocalObjectReference `json:"connectionSecretRef,omitempty"`
+	Name                string                      `json:"name,omitempty"`
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Container is the Schema for the Container
+// +k8s:openapi-gen=true
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="STORAGE_ACCOUNT",type="string",JSONPath=".spec.accountRef.name"
+// +kubebuilder:printcolumn:name="PUBLIC_ACCESS_TYPE",type="string",JSONPath=".spec.publicAccessType"
+// +kubebuilder:printcolumn:name="CLASS",type="string",JSONPath=".spec.classRef.name"
+// +kubebuilder:printcolumn:name="RECLAIM_POLICY",type="string",JSONPath=".spec.reclaimPolicy"
+// +kubebuilder:printcolumn:name="AGE",type="date",JSONPath=".metadata.creationTimestamp"
+type Container struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              ContainerSpec   `json:"spec,omitempty"`
+	Status            ContainerStatus `json:"status,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ContainerList - list of the container objects
+type ContainerList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []Container `json:"items"`
+}
+
+// GetContainerName based on the NameFormat spec value,
+// If name format is not provided, container name defaults to UID
+// If name format provided with '%s' value, container name will result in formatted string + UID,
+//   NOTE: only single %s substitution is supported
+// If name format does not contain '%s' substitution, i.e. a constant string, the
+// constant string value is returned back
+//
+// Examples:
+//   For all examples assume "UID" = "test-uid"
+//   1. NameFormat = "", ContainerName = "test-uid"
+//   2. NameFormat = "foo", ContainerName = "foo"
+//   3. NameFormat = "foo-%s", ContainerName = "foo-test-uid"
+//   4. NameFormat = "foo-%s-bar-%s", ContainerName = "foo-test-uild-bar-%!s(MISSING)"
+func (c *Container) GetContainerName() string {
```

Comment:
what scenarios is it important to give the user the ability to influence the container name by allowing them to specify a format string?  most of our other types don't allow that, is there a special use case for this with azure blob storage that makes it necessary?

Reply:
This gives users flexibility to create resource with the specific name.